ruby/ql/lib/codeql/ruby/typetracking/TypeTrackerSpecific.qll

```diff
@@ -80,10 +80,8 @@ predicate jumpStep = DataFlowPrivate::jumpStep/2;
 pragma[nomagic]
 private predicate flowThrough(DataFlowPublic::ParameterNode param) {
   exists(DataFlowPrivate::ReturningNode returnNode, DataFlowDispatch::ReturnKind rk |
-    DataFlowPrivate::LocalFlow::getParameterDefNode(param.getParameter())
-        .(TypeTrackingNode)
-        .flowsTo(returnNode) and
-    rk = returnNode.getKind()
```

Comment:
Wasn't there a bug here since the return node was not previously checked to be in the same CFG scope as the parameter? How come we don't lose any (spurious) call edges from fixing this? `flowsTo` includes some steps into closures so it should be possible to reach the wrong return node.

Reply:
It is only possible for `flowsTo` to switch CFG scope via `localFlowSsaParamCaptureInput`, and that predicate is limited to parameters and not their corresponding SSA definitions. So we were saved by the fact that `getParameterDefNode` returns the SSA definition for the parameter.